### PR TITLE
Allow skip and configuration of install dirs

### PR DESCRIPTION
**Pull request by karajas**
This change is allow the sdk repo to 
1. Allow configuration of NUGET_PACKAGES to provide an alternate cache
2. Checks if DOTNET_INSTALL_DIR exists before installing
3. Skip product build and tests if required.

### CODE REVIEW

File: build.ps1 [hunk 1]
```diff
@@ -101,17 +112,17 @@ $msbuildSummaryLog = Join-Path -path $logPath -childPath "templates.log"
 $msbuildWarningLog = Join-Path -path $logPath -childPath "templates.wrn"
 $msbuildFailureLog = Join-Path -path $logPath -childPath "templates.err"
 
-msbuild /t:restore /p:RestorePackagesPath=$PackagesPath $RepoRoot\sdk-templates.sln /verbosity:$Verbosity
+dotnet msbuild /t:restore /p:RestorePackagesPath=$PackagesPath $RepoRoot\sdk-templates.sln /verbosity:$Verbosity
```
Comment (nguerrera) [inline build.ps1]: Why did you change this?

Comment (nguerrera) [inline build.ps1]: This was done and reverted before. It broke signed builds, but I don't recall why. These are windows-only builds. I am pushing to have them moved to a different repo.

Comment (nguerrera) [inline build.ps1]: cc @srivatsn who may recall exactly what broke

**Review by nguerrera — CHANGES_REQUESTED** [review]
See above

**Comment by nguerrera** [comment]
@karajas Do you still need this?

**Comment by karajas** [comment]
At the moment, I have a workaround in the source build repo. Though I'd like to clean this up and check it in. I'll try to get to this within the next few days

**Comment by nguerrera** [comment]
@karajas I'm going to close this since there hasn't been activity in a while. Feel free to re-open or start another PR when you're ready to pick it back up.

**Pull request closed without merging**